### PR TITLE
Removes 3rd party dependencies

### PR DESCRIPTION
**Pull request by AdamPro13**

### CODE REVIEW

File: AdSpecter-iOS/AdManager+Networking.swift [hunk 1]
```diff
@@ -16,34 +15,26 @@ extension AdManager {
...
-                self.impression = updatedImpression
+            parameters: parameters
+        ) { result in
+            switch result {
+            case let .failure(error):
+                completion?(error)
+
+            case let .success(json):
+                var copiedJSON = json
+                // TODO: This value should change
```
Comment (johnli611) [inline AdSpecter-iOS/AdManager+Networking.swift]: change comment to indicate that date should be provided by server

File: AdSpecter-iOS/AdManager+Networking.swift [hunk 2]
```diff
@@ -53,58 +44,51 @@ extension AdManager {
...
-                return
-            }
-
-            guard let imageURL = URL(string: imageURLString) else {
-                completion?(APIClientError.invalidJSON)
-                return
-            }
+        // TODO: Change this path
+        APIClient.shared.makeRequest(
+            to: "test",
```
Comment (johnli611) [inline AdSpecter-iOS/AdManager+Networking.swift]: @adspecter to add new url in comment

Comment (johnli611) [reply AdSpecter-iOS/AdManager+Networking.swift]: "ad_units/default"

File: AdSpecter-iOS/ASRAdNode.swift [hunk 3]
```diff
@@ -37,4 +57,38 @@ public class ASRAdNode: SCNNode {
...
+            return
+        }
+
+        let tapLocation = sender.location(in: view)
+        let hits = view.hitTest(tapLocation, options: [.ignoreHiddenNodes: NSNumber(value: true)])
+        let wasTapped = hits.contains(where: { $0.node == self })
+        guard wasTapped else {
+            return
+        }
+
```
Comment (AdamPro13) [inline AdSpecter-iOS/ASRAdNode.swift:91]: Add TODO to make API request for ad being tapped

File: AdSpecter-iOS/AdManager.swift [hunk 4]
```diff
@@ -23,10 +22,11 @@ class AdManager {
 
     private var pendingNodes: [WeakObject<ASRAdNode>] = []
     var imageQueue: [UIImage] = []
+    var adsQueue: [ASRAdvertisement] = []
```
Comment (AdamPro13) [inline AdSpecter-iOS/AdManager.swift]: Remove this

File: AdSpecter-iOS/Models/ASRAdvertisement.swift [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    required init?(json: ASRJSONDictionary) {
+        guard let advertisementID = json["id"] as? Int else {
+            return nil
+        }
+
+        self.advertisementID = String(advertisementID)
+        isActive = json["active"] as? Bool ?? false
+        // TODO: Handle dates
+
+        if let rawCreatedAt = json["created_at"] as? String {
```
Comment (AdamPro13) [inline AdSpecter-iOS/Models/ASRAdvertisement.swift]: Remove date properties

File: AdSpecter-iOS/Models/ASRImpression.swift [hunk 6]
```diff
@@ -8,8 +8,8 @@
 
 import Foundation
 
-class ImpressionDataModel: JSONCodable {
-    var impressionID: Int?
+class ASRImpression: JSONCodable {
+    var impressionID: String?
```
Comment (AdamPro13) [inline AdSpecter-iOS/Models/ASRImpression.swift]: Make this nonnull

File: AdSpecter-iOS/Models/Thread+DateFormatter.swift [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+extension Thread {
+    static let identifier: String = UIApplication.shared.uniqueIdentifier(with: "iso8601Formatter")
+    
+    var iso8601Formatter: ISO8601DateFormatter {
+        if let formatter = threadDictionary[type(of: self).identifier] as? ISO8601DateFormatter {
+            return formatter
+        } else {
+            let formatter = ISO8601DateFormatter()
+            formatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
+            threadDictionary["iso8601Formatter"] = formatter
```
Comment (AdamPro13) [inline AdSpecter-iOS/Models/Thread+DateFormatter.swift]: Use correct identifier

**Pull request merged**